src/Check/Mongo/MongoExtendedConnectionParameters.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of the FiveLab Diagnostic package.
+ *
+ * (c) FiveLab <mail@fivelab.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types = 1);
+
+namespace FiveLab\Component\Diagnostic\Check\Mongo;
+
+/**
+ * Extended model that stores additional connection parameters for MongoDB.
+ */
+class MongoExtendedConnectionParameters
```

Comment:
This class is superfluous because username, password and db - must be in connection parameters, and collection must be in specific check, because you create the object with connection parameters and can pass this object to more checks.